jupyterlab_wipp_plugin_creator/jupyterlab_wipp_plugin_creator/handlers.py

```diff
@@ -0,0 +1,128 @@
+import json
+import os
+
+from shutil import copy2
+from wipp_client.wipp import gen_random_object_id
+
+from jupyter_server.base.handlers import APIHandler
+from jupyter_server.utils import url_path_join
+from .log import get_logger
+import tornado
+
+from jinja2 import Template
+
+logger = get_logger()
+# logger.setLevel(logging.INFO)
+
+class WippHandler(APIHandler):
+    @property
+    def wipp(self):
+        return self.settings["wipp-plugin-creator"]
+
+
+# Create wipp plugin based on user input
+class CreatePlugin(WippHandler):
+    @tornado.web.authenticated
+    def post(self):
+        """
+        POST request handler,
+
+        Creates a temp folder with random name,
+        Generates plugin.json, requirements.txt, dockerfile,
+        Copies files
+        Builds dockerimage,
+
+        Input format:
+            {
+                'formdata':
+                    {
+                    'name': '',
+                    'version': '',
+                    'title': '',
+                    'description': ''
+                    }
+                'filepaths':
+                    string[]
+            }
+
+        """
+
+        pwd = os.getcwd()
+
+        # Random ID follows MongoDB format
+        randomId = gen_random_object_id()
+
+        pluginOutputPath = os.getenv("PLUGIN_TEMP_LOCATION")
+        # if ENV exists
+        if pluginOutputPath:
+            os.chdir(pluginOutputPath)
```

Comment:
Please don't change the working directory! This interferes with the whole Jupyter process. If there's an error, it is getting stuck in the wrong place.

Better build the paths you need.
Instead
```
os.chdir(pluginOutputPath)
os.makedirs(f"{randomId}")
os.chdir(f"{randomId}")
pluginOutputPath = os.getcwd()
```
better do something like
```
pluginOutputPath = os.path.join(pluginOutputPath, f"{randomId}")
os.makedirs(pluginOutputPath)
```
and later use the absolute path when reading/writing instead of changing working directory there